### PR TITLE
Added gulp clean tasks

### PR DESCRIPTION
**Pull request by colin-marshall**
Fixes: https://github.com/olefredrik/FoundationPress/issues/680

### CODE REVIEW

**Comment by colin-marshall** [comment]
@olefredrik any idea why the build is failing? It just stopped once it hit `library/navigation.php` when running the code sniff tests.

**Comment by colin-marshall** [comment]
Closed and reopened the PR so the tests would run again, hoping that it was something wrong with Travis, but it's still not passing. I get nothing wrong with `library/navigation.php` when I run the code sniffs locally.

**Comment by olefredrik** [comment]
@colin-marshall : Seems to be an issue with Travis CI. Related: https://github.com/travis-ci/travis-ci/issues/3358 

I'll check what's going on and see how this can be fixed.

**Comment by colin-marshall** [comment]
Thanks @olefredrik! This can probably get merged in despite the build failing since none of this PR involves php.

**Comment by olefredrik** [comment]
@colin-marshall : I did some testing last night, hoping to get some answers for the Travis CI build failure. There is no way the build will pass unless I remove navigation.php. That does really make no sense to me, since the navigation.php has no errors and has previously passed the build without errors. I think it's good to have the WP Coding Standards check as part of the repository, but it's quite unfortunate if it should stop and err on files that are ok.

**Comment by colin-marshall** [comment]
@olefredrik are there any differences between the CodeSniffing done locally and when it's done on Travis CI? To see if maybe a new WPCS rule was breaking the build, I removed the local CodeSniffing, cleared the composer cache, reinstalled, and tested locally. All CodeSniffs passed with a couple warnings, none of which had to do with `navigation.php`.

**Comment by colin-marshall** [comment]
@olefredrik I have narrowed the problem down to the `foundationpress_top_bar_r()` function.

**Comment by colin-marshall** [comment]
Finally resolved @olefredrik!!! This is what I came to conclude about the build issue:

- There was a missing `@link` before the URL in the doc comments for `foundationpress_top_bar_r()` which was stalling the build
- The comments inside of the array in that function were also stalling the build

While fixing this I noticed that the local wpcs was only checking php files in the root directory of the theme, so I fixed it so that it checks php files in all directories. When running the sniffs locally it didn't stall or give errors when it hit `library/navigation.php`. No clue what was going on with Travis but at least we found what was causing it.

File: gulpfile.js [hunk 1]
```diff
@@ -223,13 +250,13 @@ gulp.task('default', ['build', 'browser-sync'], function() {
...
   // Sass Watch
-  gulp.watch(['assets/scss/**/*.scss'], ['sass'])
+  gulp.watch(['assets/scss/**/*.scss'], ['clean:css', 'sass'])
     .on('change', function(event) {
       logFileChange(event);
     });
 
   // JS Watch
-  gulp.watch(['assets/javascript/custom/**/*.js'], ['javascript', 'lint'])
+  gulp.watch(['assets/javascript/custom/**/*.js'], ['clean:javscript', 'javascript', 'lint'])
```
Comment (olefredrik) [inline gulpfile.js]: ```clean: javscript```` ? :)

**Comment by olefredrik** [comment]
Awesome, Colin!!! Thanks a lot! I'll sit down a few hours this evening and look through all recent issues. Just found a typo in the clean task I think. See comment above.

**Pull request merged**